packages/eslint-plugin/rules/api-no-slashes.js

```diff
@@ -20,36 +20,36 @@ const rule = function (context) {
         return;
       }
 
-      // API.one('foo/bad')
-      //          ^^^^^^^
       // API.all('ok').one('ok', 'foo/bad', 'ok')
-      //                          ^^^^^^^
-      const literalArgs = args.filter((arg) => arg.type === 'Literal' && arg.raw.includes('/'));
+      // ^^^
+      if ((getCallingIdentifier(node) || {}).name !== 'API') {
+        // console.log(getCallingIdentifier(callee));
+        // console.log('calling identifier not API');
+        return;
+      }
 
-      // var badvar = 'foo/bad'; API.one(badvar);
-      //                                 ^^^^^^
-      const variableArgs = args.filter((argument) => {
-        if (argument.type !== 'Identifier') {
-          return false;
+      // Get the source code and trigger the lint violation if any forward slashes are found in any of the args
+      // This isn't 100% accurate, but it's good enough.
+      const hasSlash = args.some((arg) => {
+        //  Check if the arg expression has a / anywhere, i.e.:
+        // .one(prevArg, 'foo/bar', nextarg)
+        // .one(prevArg, 'foo/' + barid, nextarg)
+        // .one(prevArg, `foo/${barid}`, nextarg)
+        const text = context.getSourceCode().getText(arg);
+        if (text.includes('/')) {
+          return true;
         }
-        const variable = getVariableInScope(context, argument);
-        const literalValue = _.get(variable, 'defs[0].node.init.raw', '');
-        return literalValue.includes('/');
+
+        // Check if the arg is a variable, and check if that variable was initialized with a slash somewhere
+        const variable = getVariableInScope(context, arg);
+        return variable && context.getSourceCode().getText(variable.init).includes('/');
```

Comment:
Does the same thing as above (like for the argument), but instead looks at variable references and their initializers